src/test/java/uk/gov/pay/directdebit/payments/model/CustomDateSerializerTest.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.pay.directdebit.payments.model;
+
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.time.ZonedDateTime;
+
+import static junit.framework.TestCase.assertEquals;
+
+
+public class CustomDateSerializerTest {
+    private CustomDateSerializer serializer;
+
+    @Before
+    public void before() {
+        serializer = new CustomDateSerializer();
+    }
+
+    @Test
+    public void shouldSerializeWithMillisecondPrecision() throws IOException {
+        ZonedDateTime testValue = ZonedDateTime.parse("2019-01-29T11:34:53.849Z");
```

Comment:
I think this would be a better test if the input date had greater than millisecond precision (e.g. 2019-01-29T11:34:53.849630917Z) but the final assertion checked that only millisecond precision was output